rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,252 @@
+* **Client**: system registering presentations on the list and reading it to discover other systems/parties.
+
+Other terms are as defined in the following specifications: "JSON Web Token (JWT)" [JWT], Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL], Verifiable Credentials Presentation Exchange [PE], Decentralized Identifiers 1.0 [DID].
+
+## 3. Service endpoint
+
+The service endpoint is an HTTP endpoint exposed by the server which hosts a list of Verifiable Presentations.
+Clients can query the list or submit a new Verifiable Presentation to it.
+All presentations MUST conform to the Presentation Definition associated with the discovery service (see Service Definition).
+
+Presentations MUST be encoded in JWT format as string ([JWT]).
+
+### 3.1 Registration
+
+Clients can publish a Verifiable Presentation to the list by sending it in an HTTP POST to the service endpoint.
+The HTTP request body MUST be a Verifiable Presentation (in JWT format). The content type of the request MUST be `application/json`.
+
+The server MUST validate the Verifiable Presentation as specified in section 4. If the validation fails, it MUST return a 400 Bad Request response.
+If the validation succeeds, the Verifiable Presentation MUST be added to the list and the server MUST return a 201 Created response.
+
+A credential subject (identified by `credentialSubject.id`) MUST NOT appear more than once on the list,
+so a new registration MUST replace the previous one from the same credential subject.
+
+An example posting a Verifiable Presentation in JWT format to the list:
+
+``http request
+POST /list HTTP/1.1
+Content-Type: application/json
+
+"eyCAFE.etc.etc"
+``
+
+The Verifiable Presentation MUST NOT be valid longer than the Verifiable Credentials it contains. 
+
+### 3.2 Reading
+
+Clients MUST read the list by sending an HTTP GET to the server's service endpoint.
+The server MUST return a 200 OK response with a JSON object containing:
+- `entries` REQUIRED. MUST contain a JSON array with zero or more presentations.
+- `tag` REQUIRED. MUST be a JSON string referring to the last entry. The tag is an opaque value; the client SHOULD NOT derive any meaning from it.
+
+The `tag` query parameter MAY be used by the client to request a delta next time it reads the list.
+If no `tag` query parameter is provided, the server MUST return the full list.
+Servers MUST only return the latest (valid) presentation per credential subject.
+
+Example:
+
+``http request
+GET /list?tag=510 HTTP/1.1
+Content-Type: application/json
+
+{
+  "entries": [
+    "ey1234.etc.etc",
+    "ey5678.etc.etc"
+  ],
+  "tag": "515",
+}
+``
+
+Clients MUST validate each presentation in the list as specified in section 4.
+If a presentation is valid, the client uses it in its system. If a presentation is not valid, it MUST be rejected.
+If one or more presentations are not valid, it SHOULD NOT reject the other presentations.
+
+### 3.3 List pruning
```

Comment:
How does this work with expired (not retracted) VPs? Do clients also have to prune their lists?